### ISSUE TITLE
[BUG] read_txt returns overlapping values with non-overlapping ranges

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#10620]
This seems to be an off-by-1 bug somewhere that returns additional records where I don't expect them:

```python
>>> import cudf
>>> from io import StringIO
>>> cudf.read_text(StringIO("abc..def..ghi.."), delimiter="..", byte_range=(0, 5))
0    abc..
1    def..
dtype: object
>>> cudf.read_text(StringIO("abc..def..ghi.."), delimiter="..", byte_range=(5, 10))
0    def..
1    ghi..
2
dtype: object
```

**Comment by shwina** [comment]
cc: @cwharris

**Comment by cwharris** [comment]
cc: @randerzander

**Comment by upsj** [comment]
This was fixed by #11371
